### ISSUE TITLE
NestJs Integration - InternalOAuthError

### ISSUE THREAD

**Issue by AdamJessop** [superfaceai/passport-twitter-oauth2#10]
Hi,

I am struggling to integrate this passport strategy with NestJs. I have tried  a number of different ways, from using the official `AuthGuard` and strategy class to calling passport.use / passport.authenticate directly with the same strategy instance.

Here is my strategy class: 

```
import { PassportStrategy } from '@nestjs/passport';
import * as TwitterStrategy from '@superfaceai/passport-twitter-oauth2';
...

@Injectable()
export class TenantTwitterAuthStrategy extends PassportStrategy(TwitterStrategy, 'twitter') {
    private readonly configService: ConfigService;

    constructor(config: ConfigService) {
        super({
            clientID: config.get('TWITTER_CLIENT_ID'),
            clientSecret: config.get('TWITTER_CLIENT_SECRET'),
            callbackURL: `${config.get('HOST')}/auth/tenant/twitter/redirect`,
            clientType: 'private',
            scope: ['tweet.read', 'offline.access'],
            passReqToCallback: true,
        });
        this.configService = config;
    }

    public validate(req: IAuthedRequest, accessToken: string, refreshToken: string, profile: TwitterStrategy.Profile) {
    
        ....
     }
```

I have attempted to debug, and I can see the the `userProfile` function is getting what looks like a valid accessToken, and the url object is: 

```
{
  href: 'https://api.twitter.com/2/users/me?user.fields=profile_image_url,url',
  origin: 'https://api.twitter.com',
  protocol: 'https:',
  username: '',
  password: '',
  host: 'api.twitter.com',
  hostname: 'api.twitter.com',
  port: '',
  pathname: '/2/users/me',
  search: '?user.fields=profile_image_url,url',
  searchParams: URLSearchParams { 'user.fields' => 'profile_image_url,url' },
  hash: ''
}
```

which looks correct. However it's throwing an `InternalOAuthError: Failed to fetch user profile`.

Which is a result of a `403 Forbidden` response coming back from the profile call.

Has anyone managed to sucessfully integrate this with NestJs. I haven't had issues with other providers or indeed the original `passport-twitter` library, however I am wanting to use the v2 API.

**Comment by janhalama** [comment]
Hi, looks like problem with scopes. Try to add `users.read` scope to list of requested scopes in the strategy constructor.

**Comment by AdamJessop** [comment]
Ah, gotcha, missed that one which is easily done. I raised a PR to have that scope as a minimum default for the strategy to work:

https://github.com/superfaceai/passport-twitter-oauth2/pull/11

**Comment by janhalama** [comment]
Released as [v1.1.0](https://github.com/superfaceai/passport-twitter-oauth2/releases/tag/v1.1.0)
